noir/noir-repo/aztec_macros/src/transforms/functions.rs

```diff
@@ -275,6 +283,31 @@ fn create_mark_as_initialized(ty: &str) -> Statement {
     )))
 }
 
+/// Forces a static context for a function, ensuring that no state modifications are allowed
+///
+/// ```noir
+/// assert(context.inputs.call_context.is_static_call == true,  "Function can only be called statically")
+/// ```
+fn create_static_check(fname: &str, is_avm: bool) -> Statement {
+    let is_static_call_expr = if !is_avm {
+        ["inputs", "call_context", "is_static_call"]
+            .iter()
+            .fold(variable("context"), |acc, member| member_access(acc, member))
+    } else {
+        ["inputs", "is_static_call"]
+            .iter()
+            .fold(variable("context"), |acc, member| member_access(acc, member))
+    };
+    make_statement(StatementKind::Constrain(ConstrainStatement(
+        make_eq(is_static_call_expr, expression(ExpressionKind::Literal(Literal::Bool(true)))),
+        Some(expression(ExpressionKind::Literal(Literal::Str(format!(
+            "Function {} can only be called statically",
+            fname
+        ))))),
+        ConstrainKind::Assert,
```

Comment:
Isn't it easier to use the parser for this?

Reply:
It is, but this particular section is:

1. Old, as it's the function transformer. We don't really want to invest a lot of effort migrating if the metaprogramming is coming soon ™️ 
2. Annoying, since it's a *modification* to existing code. We are injecting "lines" in the AST as opposed to generating functions from scratch.